### PR TITLE
Fix extraction issues

### PR DESCRIPTION
**Pull request by abhishekvm**
----
CRAW-208: http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615

Before:
"author": "",
"author_description": "",

Now:
"author": "The Associated Press",
"author_description": "THE ASSOCIATED PRESS",

----
CRAW-203: https://www.ft.com/content/d0a5c460-5044-11e7-a1f2-db19572361bb

Before:
"author":"",
"author_description":"",

Now:
"author": "Michael Power"
"author_description": "by: Michael Power, Investec"

----
CRAW-211: http://sdn.cioreview.com/cxoinsight/sdn-do-you-really-need-it-nid-24422-cid-147.html

Before:
"author": "CIO Review"
"author_description": "CIO Review"

Now:
"author": "Patrick Hubbard"
"author_description": "Patrick Hubbard, Head Geek, SolarWinds"

----
CRAW-212: http://www.philly.com/archive/john_timpane/Princetons-Tracy-K-Smith-named-US-Poet-Laureate.html

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

----
CRAW-210: http://nation.foxnews.com/2017/06/13/sessions-inquisition-comes-empty-trump-unscathed-after-attorney-generals-testimony

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

Before:
"author": ""
"author_description": ""

Now:
"author": "Hans A. von Spakovsky | Fox Opinion"
"author_description": "Hans A. von Spakovsky | Fox Opinion"

### CODE REVIEW

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 1]
```diff
@@ -2481,6 +2507,12 @@ protected String extractAuthorDescription(Document doc, String authorName){
             return SHelper.innerTrim(matches.first().attr("href"));
         }
 
+        // http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615
+        matches = doc.select("div[class=ra-credits]");
+        if (matches == null || matches.size() > 0){
+            return SHelper.innerTrim(matches.first().ownText());
```
Comment (andresp99999) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: Maybe add a DEBUG_AUTHOR_EXTRACTION log here?

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 2]
```diff
@@ -2203,6 +2217,18 @@ protected String extractAuthorName(Document doc) {
...
 
+                    // http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615
+                    if (matches == null || matches.size() == 0){
+                        matches = doc.select("div[itemtype$=schema.org/Person]");
+                        if(DEBUG_AUTHOR_EXTRACTION && matches!=null && matches.size()>0) System.out.println("AUTHOR: div[itemtype$=schema.org/Person]");
+                    }
+
+                    // http://nation.foxnews.com/2017/06/13/sessions-inquisition-comes-empty-trump-unscathed-after-attorney-generals-testimony
+                    if (matches == null || matches.size() == 0){
+                        matches = doc.select("p > strong:contains(By)");
```
Comment (skyshard) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: This looks like it would be a slow selector

**Review by skyshard — CHANGES_REQUESTED** [review]

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java (hunk 2 above)
Comment (abhishekvm) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: @skyshard  The article uses pretty much plain html tags without distinguishing css.

Comment (abhishekvm) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: @skyshard - Can you confirm what can be done here? Should we merge this as is or remove fix for nation.foxnews.com?

Comment (andresp99999) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: @abhishek-airpr let's remove the nation.foxnews.com hack for now, after that feel free to merge it.

**Pull request merged**